src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
+  private logger = new Logger();
+
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler,
+  ): Observable<Response<T>> {
+    const statusCode = context.switchToHttp().getResponse().statusCode;
+    const req = context.switchToHttp().getRequest();
+    const userDetail = req?.user || {};
+
+    return next.handle().pipe(
+      map((data) => {
+        const response = {
+          isSuccess: statusCode >= 200 && statusCode < 300 ? true : false,
+          statusCode: statusCode,
+          error: data && data.message ? data.message : '',
+          message: data && data.message ? data.message : '',
+          data: data && data.message ? { ...data, message: undefined } : data,
```

Comment:
PullProofAI

The `message` field is being set twice based on the same condition. It can be optimized to set it once.
```diff
-           message: data && data.message ? data.message : '',
-           data: data && data.message ? { ...data, message: undefined } : data,
+           message: data && data.message ? data.message : '', // Keep only one assignment
+           data: data && data.message ? { ...data, message: undefined } : data,
```


<!-- This is an auto-generated comment by PullProofAI -->